sbin/repair_services.sh

```diff
@@ -1,36 +1,121 @@
 #!/usr/bin/env bash
 # shellcheck enable=all
 
-#### This script repairs core services that are not currently running
+#### This script repairs core services that are not currently active
 
-repair_services() {
-    services=(
-        "nginx"
-        "apache2"
-        "postgresql"
-        "nullsmtpd"
-        "submitty_websocket_server"
-        "submitty_autograding_worker"
-        "submitty_autograding_shipper"
-        "submitty_daemon_jobs_handler"
-    )
-
-    for service in "${services[@]}"; do
-        if ! sudo systemctl is-active --quiet "${service}"; then
-            today=$(date +%Y%m%d)
-            log_file="/var/log/services/${today}.txt"
-            last_status=$(sudo systemctl status "${service}")
-
-            if [[ ! -f "${log_file}" ]]; then
-                sudo touch "${log_file}"
+declare -A services=(
+    ["nginx"]="Nginx Server"
+    ["apache2"]="Web Server"
+    ["postgresql"]="Database Server"
+    ["autograding"]="Autograding Shipper/Workers"
+    ["submitty_websocket_server"]="WebSocket Server"
+    ["submitty_daemon_jobs_handler"]="Daemon Jobs Handler"
+)
+
+log_service_restart() {
+    local service="$1"
+    local message="$2"
+    local last_status="$3"
+
+    local today
+    today=$(date +%Y%m%d)
+    local timestamp
+    timestamp=$(date "+%Y-%m-%d:%H:%M:%S")
+    local log_file="/var/local/submitty/logs/services/${today}.txt"
+
+    if [[ ! -f "${log_file}" ]]; then
+        sudo touch "${log_file}"
+    fi
+
+    echo -e "${timestamp}: ${services["${service}"]}\n" | sudo tee -a "${log_file}"
+    echo -e "${message}\n\n${last_status}" | sudo tee -a "${log_file}"
+    echo -e "----------------------------------------" | sudo tee -a "${log_file}"
+}
+
+repair_autograding() {
```

Comment:
Rather than duplicate logic...  I would rather just call the existing script:

 sbin/restart_shipper_and_all_workers.py 
 
If any problems are detected with the shippers or workers on the primary machine.
This halts all all the shippers on the primary machine AND the workers on the primary machine AND the workers processes on worker machines AND brings up all the shippers & workers in the proper order.   If we change anything in the future about how these work, we don't want to have to (remember to) make the update in 2 places.